### ISSUE TITLE
Minimum required version for Ruby 3.1

### ISSUE THREAD

**Issue by zacccypoo** [restforce/restforce#868]
### Summary

Hello, I apologize if this is the wrong forum for this question. I have been tasked with upgrading my company's application from Ruby 2.7 -> 3.1. We rely heavily on the restforce gem and while it is tremendously outdated, I am having trouble understanding if our old version is supported in Ruby 3

- Currently we are on restforce 4.2.2
- The [Changelog](https://github.com/restforce/restforce/blob/main/CHANGELOG.md#523-jan-17-2022) says you officially support Ruby 3.1 in in restforce 5.2.3
- The [Rubygems page for 5.2.3](https://rubygems.org/gems/restforce/versions/5.2.3) says the minimum version of ruby required is >= 2.6
- The [Rubygems page for 4.2.2](https://rubygems.org/gems/restforce/versions/4.2.2) says the minimum version of ruby required is >= 2.4
  - but nowhere does it say the maximum version of ruby supported 
- The [README](https://github.com/restforce/restforce/blob/main/README.md#installation) and the [Rubygems page for 7.0.0](https://rubygems.org/gems/restforce/versions/7.0.0) both say the minimum for the 7.0 is only supported in Ruby >= 3.0
- Given some strict time constraints I am hoping to only upgrade the bare minimum of any dependency

From what I have read, Ruby 3 introduced a bunch of breaking changes to the language itself. Are there breaking changes that would make restforce 4.2.2 incompatible with ruby 3? I guess more generally speaking, how would I determine the maximum ruby version supported by the different versions of your gem.

Feel free to close this and tell me to go away but if you have any advice it would be a tremendous help.

**Comment by timrogers** [comment]
My recommendation would be to try running the automated tests for Restforce 4.2.2 with Ruby 3.x. If that works, then I'd have a decent level of confidence that everything will be fine.

**Comment by zacccypoo** [comment]
Thanks for the quick response @timrogers i really appreciate it.

I did a bunch of testing on different ruby versions / restforce versions yesterday. 

- For one reason or another Restforce 4.2.2 with Ruby 3.1 yields failing tests in my environment.
- Strange thing for me is I see (less) failing tests with Restforce 4.2.2 with Ruby 2.7.8
- Restforce 5.2.1 (in the changelog says officially supports ruby 3.1) i see 2 tests fail
- Restforce 5.3.1 with Ruby 3.1 is all green for me

I wouldnt be surprised if I am doing something wrong here. The big change is in Faraday more than restforce. Given that Faraday seems to also require an upgrade we have decided it is safest to upgrade faraday and everything that depends on it like restforce.

So I will be bumping to Restforce 5.3.1, thank you for the advice. We will likely audit our other gems and make sure the unit tests are all passing for our selected versions
